contracts/samples/SimpleAccountFactory.sol

```diff
@@ -26,27 +27,35 @@ contract SimpleAccountFactory {
      * This method returns an existing account address so that entryPoint.getSenderAddress() would work even after account creation
      */
     function createAccount(address owner,uint256 salt) public returns (SimpleAccount ret) {
-        address addr = getAddress(owner, salt);
-        uint codeSize = addr.code.length;
```

Comment:
the real gas cost saving here is not the `addr.code.length` (which indeed, is only 100, since the address will get warm), but the getAddress: there is no need to calculate the full create2 address (including bringing into memory the entire initCode and hash it)
Since we only need the address calculation if the account is not deployed, it makes sense to make it only after create2 marks that the account is already deployed (which can only happen if the method is used off-chain.